pkg/resource/deploy/deployment.go

```diff
@@ -249,6 +261,10 @@ func (m *resourcePlans) plan() *Plan {
 type Deployment struct {
 	// the plugin context (for provider operations).
 	ctx *plugin.Context
+	// options for this deployment.
+	opts *Options
```

Comment:
I've made this a pointer to be consistent (and because the struct is not super tiny), but ideally we'd have some way to convey that we _really_ don't want you to be mutating this through the deployment.